[IMP] helpdesk: updated 16.0 reporting doc (#3690)
odoo/documentation: content/applications/services/helpdesk/overview/reports.rst

=========
Reporting
=========

Odoo :guilabel:`Helpdesk` features several reports that provide the opportunity to track trends,

> identify areas for improvement, manage employees` workloads, and confirm when customer expectations

Reviewer: ```suggestion
identify areas for improvement, manage employee workloads, and confirm when customer expectations
```

are met.

Available reports
=================

> The reports below can be found under :menuselection:`Helpdesk --> Reporting`.

Reviewer: ```suggestion
The following are the reports available in Odoo :guilabel:`Helpdesk`. To view the different reports, go to :menuselection:`Helpdesk --> Reporting`.
```

Tickets Analysis
----------------

> The :guilabel:`Tickets Analysis` report provides an overview of every ticket in the database. The

Reviewer: ```suggestion
The :guilabel:`Tickets Analysis` report provides an overview of every customer support ticket in the database. The
```

default report counts the number of tickets per team and groups them by stage.

.. image:: reports/reports-tickets-default.png
   :align: center
   :alt: View of Tickets Analysis report default view

Alternative measures can be selected to track where the most time is spent at different points in
the workflow. To change the metric used in the current report, choose an option from the

> :guilabel:`Measures` dropdown.

Reviewer: Same as below - is it worth mentioning some of the different Measures available?

.. image:: reports/reports-tickets-additionalmeasures.png
   :align: center
   :alt: View of additional measures on Tickets Analysis report

To sort tickets by the user assigned to them, select :menuselection:`Group by --> Assigned to`.

> Additional criteria can be selected from the :guilabel:`Group By` dropdown in order to get a more

Reviewer: Is it worth listing/explaining the different Group By functions, like different criteria that you can set? Even just a quick sentence about each

detailed overview of the current workload of every member of the support team.

.. image:: reports/reports-tickets-assignedto.png
   :align: center
   :alt: View of Group by options of Tickets Analysis report

SLA Status Analysis
-------------------

> The :guilabel:`SLA Status Analysis` report tracks how quickly an SLA is fulfilled, as well as the

Reviewer: ```suggestion
The :guilabel:`SLA Status Analysis` report tracks how quickly an :abbr:`SLA (Service Level Agreement` is fulfilled, as well as the
```

> success rate of individual policies. By default, the report is filtered to show the number of SLAs

Reviewer: ```suggestion
success rate of individual policies. By default, the report is filtered to show the number of :abbr:`SLAs (Service Level Agreements)`
```

> failed and the failure rate over the last 30 days, grouped by team.

Reviewer: ```suggestion
failed, as well as the failure rate over the last 30 days, grouped by team.
```

.. image:: reports/reports-sla-status.png
   :align: center
   :alt: View of Group by options of Tickets Analysis report

> To add additional measures, click :guilabel:`Measures` and choose and option from the drop down. To

Reviewer: ```suggestion
To add additional measures, click :guilabel:`Measures` and choose an option from the drop-down menu. To
```

add an additional group at the row level, click :guilabel:`Total`.

.. example::

>    To see the number of tickets that were able to achieve the stated SLA objectives, and track the

Reviewer: ```suggestion
   To see the number of tickets that were able to achieve the stated :abbr:`SLA (Service Level Agreement)` objectives, and track the
```

   amount of time it took to achieve those objectives, click
   :menuselection:`Measures --> Number of SLA Successful` and
   :menuselection:`Measures --> Workings Hours to Reach SLA`.
   To sort these results by the team members assigned to the tickets, select
   :menuselection:`Total --> Assigned to`.

Customer Ratings
----------------

The :guilabel:`Customer Ratings` report displays an overview of the ratings received on individual

> tickets, as well as any additional comments submitted with the rating.

Reviewer: ```suggestion
support tickets, as well as any additional comments submitted with the rating.
```

.. image:: reports/reports-customer-ratings.png
   :align: center
   :alt: View of the kanban display in the Customer Ratings report

> Click on an individual rating to see additional details, including a link to the original ticket.

Reviewer: ```suggestion
Click on an individual rating to see additional details about the rating assigned by the customer, including a link to the original ticket.
```

.. image:: reports/reports-ratings-details.png
   :align: center
   :alt: View of the details of an individual customer rating

.. tip::
   Select :guilabel:`Visible Internally Only` to hide a rating from the customer portal.

Save filters
============

The :guilabel:`Favorites` feature allows users to save the most commonly used filters without
having to reconstruct them every time they are needed.

> To create a new favorite:

Reviewer: ```suggestion
To create a new :guilabel:`Favorite`:
```

=========
Reporting
=========

Odoo's :guilabel:`Helpdesk` app includes several reports that provide the opportunity to track
trends for customer support tickets, identify areas for improvement, manage employee workloads,
and confirm when customer expectations are met.

Available reports
=================

The following are the reports available in Odoo :guilabel:`Helpdesk`. To view the different reports,
go to :menuselection:`Helpdesk --> Reporting`.

Ticket Analysis
---------------

> The :guilabel:`Ticket Analysis` report provides an overview of every customer support ticket in the
> database, including the number of tickets assigned among teams and individual users. This report is
> useful in identifying where teams are spending the most time, and determining if there is an uneven
> workload distribution among the support staff.

Reviewer 2: ```suggestion
The :guilabel:`Ticket Analysis` reporting page (:menuselection:`Helpdesk app --> Reporting --> Ticket Analysis`) provides an overview of every customer support ticket in the database, including the number of tickets assigned among teams and individual users.

This report is useful in identifying where teams are spending the most time, and helps determine if there is an uneven workload distribution among the support staff.
```

The default report counts the number of tickets per team and groups them by stage.

.. image:: reports/reports-tickets-default.png
   :align: center
   :alt: View of Ticket Analysis report default view

Alternative measures can be selected to track where the most time is spent at different points in

> the workflow. To change the measures used in the report, click :guilabel:`Measures` and choose from
> the drop-down menu. The available measures for the :guilabel:`Ticket Analysis` report include:

Reviewer 2: ```suggestion
the workflow. To change the measures used in the report, click the :guilabel:`Measures` button, and choose an option from the drop-down menu. The available measures for the :guilabel:`Ticket Analysis` report include:
```

- Average Hours to Respond
- Hours Open
- Hours Spent
- Hours to Assign
- Hours to Close
- Hours to First Response
- Hours to SLA Deadline
- Rating /5

> - Count

Reviewer 2: Since these are menu options, I believe the correct protocol would be to give each of these :guilabels:

Also, it might be beneficial if there was a screenshot taken of the open Measures menu, showing all these options, where the Measures menu is on the screen, and so on...

Just a thought...

Author: confirmed with sam, since it isn't a direction/command, they don't get guilabels, but they can be italicized, so I made that update👍

Reviewer 3: @jero-odoo KC's suggestion is correct because this this is a UI walk-through that is a continuation of an active instruction (e.g. open the menu and then make a choice).

Here's a good example of what that looks like, where each UI element is given a guilabel and a small description after a colon, like this: https://github.com/odoo/documentation/pull/3532

To force the guilabel use, I would remove the previous sentence before the list and end the previous one with a semicolon:
```
To change the measures used in the report, click the :guilabel:`Measures` button, and choose an option from the drop-down menu:
```

SLA Status Analysis
-------------------

> The :guilabel:`SLA Status Analysis` report tracks how quickly an :abbr:`SLA (Service Level
> Agreement` is fulfilled, as well as the success rate of individual policies. By default, the report
> is filtered to show the number of :abbr:`SLAs (Service Level Agreements)` failed, as well as the
> failure rate over the last 30 days, grouped by team.

Reviewer 2: ```suggestion
The :guilabel:`SLA Status Analysis` reporting page (:menuselection:`Helpdesk app --> Reporting --> SLA Status Analysis) tracks how quickly an :abbr:`SLA (Service Level Agreement)` is fulfilled, as well as the success rate of individual policies.

By default, this report is filtered to show the number of :abbr:`SLAs (Service Level Agreements)` failed, as well as the
failure rate over the last 30 days, grouped by team.
```

.. image:: reports/reports-sla-status.png
   :align: center
   :alt: View of Group by options of Ticket Analysis report

> To change the measures used in the report, click :guilabel:`Measures` and choose from the drop-down
> menu. The available measures for the :guilabel:`SLA Status Analysis` report include:

Reviewer 2: ```suggestion
To change the measures used in the report, click the :guilabel:`Measures` button, and choose an option from the drop-down menu. The available measures for the :guilabel:`SLA Status Analysis` report include:
```

- % of Failed SLA
- % of SLA in Progress
- % of Successful SLA
- Number of SLA Failed
- Number of SLA Successful
- Number of SLA in Progress
- Working Hours to Assign
- Working Hours to Close
- Working to Reach SLA

> - Count

Reviewer 2: Same note as above

.. example::

>    To see the number of tickets that were able to achieve the stated :abbr:`SLA (Service Level
>    Agreement)` objectives, and track the amount of time it took to achieve those objectives, click
>    :menuselection:`Measures --> Number of SLA Successful` and :menuselection:`Measures --> Workings
>    Hours to Reach SLA`. To sort these results by the team members assigned to the tickets, select
>    :menuselection:`Total --> Assigned to`.

Reviewer 2: ```suggestion
   To see the number of tickets that were able to achieve the stated :abbr:`SLA (Service Level
   Agreement)` objectives, and track the amount of time it took to achieve those objectives, click :menuselection:`Measures --> Number of SLA Successful`, and then :menuselection:`Measures --> Workings Hours to Reach SLA`. 
   
   To sort these results by the team members assigned to the tickets, select :menuselection:`Total --> Assigned to`.
```

.. seealso::
   - :doc:`sla`

Customer Ratings
----------------

> The :guilabel:`Customer Ratings` report displays an overview of the ratings received on individual
> support tickets, as well as any additional comments submitted with the rating.

Reviewer 2: ```suggestion
The :guilabel:`Customer Ratings` report page (:menuselection:`Helpdesk app --> Reporting --> Customer Ratings`) displays an overview of the ratings received on individual support tickets, as well as any additional comments submitted with the rating.
```

.. image:: reports/reports-customer-ratings.png
   :align: center
   :alt: View of the kanban display in the Customer Ratings report

Click on an individual rating to see additional details about the rating assigned by the customer,
including a link to the original ticket.

.. image:: reports/reports-ratings-details.png
   :align: center
   :alt: View of the details of an individual customer rating

.. tip::

>    Select :guilabel:`Visible Internally Only` to hide a rating from the customer portal.

Reviewer 2: ```suggestion
   On a specific rating page, select the :guilabel:`Visible Internally Only` option to hide that rating from the customer portal.
```

> The :guilabel:`Customer Ratings` report is displayed in a kanban view by default, but can also be
> displayed in graph or pivot view.

Reviewer 2: ```suggestion
The :guilabel:`Customer Ratings` report page is displayed in a kanban view by default, but can also be
displayed in graph, list, or pivot view.
```

Unsure if those view options should have a :guilabel:, since they are, technically, visible/clickable options on the page, itself. Might be worth looking into.

Author: confirmed with @samueljlieber they do not need a guilabel 👍

.. seealso::
   - :doc:`ratings`

> Choosing the view and filters

Reviewer 2: View and filter options

=============================

> The best view to display a report will depend on the what data is being measured, and how it needs
> to be grouped. See below for additional information on the available views for the
> :guilabel:`Helpdesk` reports.

Reviewer 2: ```suggestion
On any :guilabel:`Reporting` page, the ideal view and filter options vary, depending on what data is currently being analyzed, measured, and grouped. See below for additional information on the available views for the various :guilabel:`Helpdesk` reports.
```

.. note::

>    Only one measure may be selected at a time for graphs, while pivot tables can include multiple
>    measures.

Reviewer 2: ```suggestion
   Only one measure may be selected at a time for graphs, but pivot tables can include multiple measures.
```

> Using the pivot view

Reviewer 2: Pivot view

--------------------

> The *pivot* view presents data in an interactive manner. All three :guilabel:`Helpdesk` reports are
> available in the pivot view.

Reviewer 2: ```suggestion
The *Pivot* view presents data in an interactive manner. All three :guilabel:`Helpdesk` reports are
available in pivot view.
```

> Switch a report to the pivot view by selecting the icon at the top right of the screen.

Reviewer 2: ```suggestion
On any :guilabel:`Reporting` page, the pivot view can be accessed, by selecting the 'grid' icon at the top right of the screen, beneath the search bar.
```

.. image:: reports/reports-pivot-view.png
   :align: center
   :alt: View of the SLA status analysis report in Odoo Helpdesk

> To add a group to a row or column, click the plus button (:guilabel:`+`) next to :guilabel:`Total`,
> and then select one of the groups. To remove one, click the minus button (:guilabel:`-`) and
> de-select.

Reviewer 2: ```suggestion
To add a group to a row or column to the pivot view, click the plus button (:guilabel:`+`) next to :guilabel:`Total`,
and select one of the groups. To remove one, click the minus button (:guilabel:`-`) and de-select the desired option.
```

> Using the graph view

Reviewer 2: Graph view

--------------------

> The *graph* view presents data in either a *bar*, *line*, or *pie* chart. Switch to the graph view
> by selecting the icon at the top right of the screen. To switch between the different chart views,
> select the appropriate icon at the top left of the chart.

Reviewer 2: ```suggestion
The *Graph* view option presents data in either a *bar* graph, *line* graph, or *pie* chart. 

Switch to the graph view by selecting the 'line graph' icon at the top right of the screen, beneath the search bar. To switch between the different chart views, select the desired graph-related icon at the top left of the chart, while in graph view.
```

.. tabs::

>    .. tab:: Bar chart

Reviewer 2: ```suggestion
   .. tab:: Bar graph
```

Author: I know it seems counterintuitive (since it is the "graph view") but when you hover over the individual icons for each of these (bar, line, pie) they are referred to as "charts" and not graphs. I realize it might not seem correct, however since it follows the UI, I'd rather match that. (I included the labels in the screenshots for that reason as well).

If it seems out of place I don't mind changing it, but I wanted to match the UI

       .. image:: reports/reports-bar-chart.png
          :align: center
          :alt: View of the SLA status analysis report in bar view

>    .. tab:: Line chart

Reviewer 2: ```suggestion
   .. tab:: Line graph
```

Author: see above

       .. image:: reports/reports-line-chart.png
          :align: center
          :alt: View of the Customer Ratings report in line view

   .. tab:: Pie chart

       .. image:: reports/reports-pie-chart.png
          :align: center
          :alt: View of the Ticket analysis report in pie chart view

.. tip::

>    Both the :guilabel:`Bar Chart` and :guilabel:`Line Chart` can be viewed :guilabel:`Stacked`.
>    This presents two or more groups to appear on top of each other instead of next to each other,
>    making it easier to compare data.

Reviewer 2: ```suggestion
   The *Bar* and *Line* graphs can be utilize the :guilabel:`Stacked` option.
   This options presents two (or more) groups on top of each other, instead of next to each other,
   making it easier to compare data.
```

Save and share a favorite search
--------------------------------

> The :guilabel:`Favorites` feature allows users to save their most commonly used filters without
> having to reconstruct them every time they are needed.

Reviewer 2: ```suggestion
The :guilabel:`Favorites` feature on the various :guilabel:`Reporting` pages in the *Helpdesk* app allows users to save their most commonly used filters, without having to reconstruct them.
```

> To create and save new :guilabel:`Favorites`:

Reviewer 2: ```suggestion
To create and save new :guilabel:`Favorites` on a :guilabel:`Repoting` page, follow the steps below:
```

> - Set the necessary parameters using the :guilabel:`Filters`, :guilabel:`Group By` and
>   :guilabel:`Measures`.

Reviewer 2: ```suggestion
- First, set the necessary parameters using the :guilabel:`Filters`, :guilabel:`Group By` and
  :guilabel:`Measures` options.
```

- Click :menuselection:`Favorites --> Save current search`.
- Rename the search.
- Select :guilabel:`Use by default` to have these filter settings automatically displayed when
  the report is opened. Otherwise, leave it blank.
- Select :guilabel:`Share with all users` to make this filter available to all other database
  users. If this box is not checked, it will only be available to the user who creates it.

> - Click :guilabel:`Save`.

Reviewer 2: ```suggestion
- Click :guilabel:`Save` to save the filter for future use.
```

=========
Reporting
=========

Odoo *Helpdesk* includes several reports that provide the opportunity to track trends for customer
support tickets, identify areas for improvement, manage employee workloads, and confirm when
customer expectations are met.

Available reports
=================

Details about the reports available in Odoo *Helpdesk* can be found below. To view the different
reports, go to :menuselection:`Helpdesk --> Reporting`.

Ticket Analysis
---------------

The *Ticket Analysis* report (:menuselection:`Helpdesk --> Reporting --> Ticket Analysis`) provides
an overview of every customer support ticket in the database. This includes the number of tickets
assigned among teams and individual users.

This report is useful in identifying where teams are spending the most time, and helps determine if
there is an uneven workload distribution among the support staff. The default report counts the
number of tickets per team and groups them by stage.

> .. image:: reports/reports-tickets-default.png

Reviewer 4: All images: we don't need to prepend `reports-` to the image filename since it is already implied by residing in the `reports/` folder

   :align: center
   :alt: View of Ticket Analysis report default view.

Alternative measures can be selected to track where the most time is spent at different points in
the workflow. To change the measures used for the report that is currently displayed, or to add
more, click the :guilabel:`Measures` button, and select one or more options from the drop-down
menu:

- :guilabel:`Average Hours to Respond`
- :guilabel:`Hours Open`
- :guilabel:`Hours Spent`
- :guilabel:`Hours to Assign`
- :guilabel:`Hours to Close`
- :guilabel:`Hours to First Response`
- :guilabel:`Hours to SLA Deadline`
- :guilabel:`Rating /5`
- :guilabel:`Count`

SLA Status Analysis
-------------------

> The *SLA Status Analysis* report (:menuselection:`Helpdesk --> Reporting --> SLA Status Analysis)`

Reviewer 4: pull the closing `)` out of the :menuselection:

failed, as well as the failure rate over the last 30 days, grouped by team.

.. image:: reports/reports-sla-status.png
   :align: center
   :alt: View of Group by options of Ticket Analysis report.

To change the measures used for the report that is currently displayed, or to add more, click the
:guilabel:`Measures` button, and select one or more options from the drop-down menu:

- :guilabel:`% of Failed SLA`
- :guilabel:`% of SLA in Progress`
- :guilabel:`% of Successful SLA`
- :guilabel:`Number of SLA Failed`
- :guilabel:`Number of SLA Successful`
- :guilabel:`Number of SLA in Progress`
- :guilabel:`Working Hours to Assign`
- :guilabel:`Working Hours to Close`
- :guilabel:`Working to Reach SLA`
- :guilabel:`Count`

.. example::
   To see the number of tickets that were able to achieve the stated :abbr:`SLA (Service Level
   Agreement)` objectives, and track the amount of time it took to achieve those objectives, click
   :menuselection:`Measures --> Number of SLA Successful` and :menuselection:`Measures --> Workings
   Hours to Reach SLA`.

   To sort these results by the team members assigned to the tickets, select

>    :menuselection:`Total --> Assigned to`.

Reviewer 4: can break this line within the :menuselection: anywhere there is a space

   .. tab:: Line chart

       .. image:: reports/reports-line-chart.png
          :align: center
          :alt: View of the Customer Ratings report in line view.

   .. tab:: Pie chart

       .. image:: reports/reports-pie-chart.png
          :align: center
          :alt: View of the Ticket analysis report in pie chart view.

.. tip::
   Both the *bar chart* and *line chart* can utilize the *stacked* view option. This presents two
   (or more) groups of data on top of each other, instead of next to each other, making it easier
   to compare data.

Save and share a favorite search
--------------------------------

The *Favorites* feature found on *Helpdesk* reports allows users to save their most commonly used
filters without having to reconstruct them every time they are needed.

To create and save new *Favorites* on a report, follow the steps below:

- Set the necessary parameters using the :guilabel:`Filters`, :guilabel:`Group By` and
  :guilabel:`Measures` options.
- Click :menuselection:`Favorites --> Save current search`.
- Rename the search.
- Select :guilabel:`Use by default` to have these filter settings automatically displayed when

>   the report is opened. Otherwise, leave it blank.

Reviewer 4: can break line later
